scripts/hook.py

```diff
@@ -160,6 +162,7 @@ def forward(self, x, timesteps=None, context=None, **kwargs):
                 # hires stuffs
                 # note that this method may not works if hr_scale < 1.1
                 if abs(x.shape[-1] - param.hint_cond.shape[-1] // 8) > 8:
+                    param.hint_cond = param.hr_hint_cond
```

Comment:
Just tested it, and it does seem to be a big problem. It shouldn't be too difficult to fix for other extensions, they simply have to also overwrite `param.hr_hint_cond`.
The only fix I see working is to pass an additional copy of `hint_cond`, and only switch to `hr_hint_cond` if the copy and the current `hint_cond`match

Reply:
Adding a @property decorator and setting it to update hr_hint_cont should maintain the previous behaviour of updating `hint_cond`, while also allowing extensions to modify `hr_hint_cond`. Something like adding this to the ControlParams class:

```python
    @property
    def hint_cond(self):
        return self._hint_cond

    # this will also be called when setting it to the hr version, which may pose other problems,
    # but I don't know yet
    @hint_cond.setter
    def hint_cond(self, new_hint_cond):
        self._hint_cond = new_hint_cond
        self.hr_hint_cond = new_hint_cond
```

This lets me remove the comparison with the original, and is the least intrusive way that I see of implementing this that doesn't break anything and allows expansion by other scripts. I don't really know the all the extensions and stuff well enough to write the callback stuff or modify the API.

Reply:
I added the @property, should now be compatible with all extensions without breaking anything.